DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/user/user.service.js

```diff
@@ -0,0 +1,35 @@
+﻿(function() {
+    'use strict';
+    angular
+        .module('app.user')
+        .factory('userService', userService);
+
+    userService.$inject = ['$cookies', '$http', 'API_URL'];
+
+    /* @ngInject */
+    function userService($cookies, $http, API_URL) {
+        function login() {
+            location.href = API_URL.LOGIN;
+        }
+
+        function logout() {
+            $cookies.remove('userName');
+            $http.post(API_URL.LOGOUT, {});
+        }
+
+        function userName() {
+            return $cookies.get('userName');
+        }
+
+        function isAuthenticated() {
+            return !!$cookies.get('userName');
```

Comment:
I think vulnerabilities of "isAuth" and "userName" cookies on frontend are the same. So it's important to be sure that we cannot get empty user's name from backend to use !!$cookies.get('userName');
I will check if it possible, but at this moment I restored "isAuth".